[stable/datadog] KSM update and add doc for secret creation

clamoriniere opened the pull request:
#### What this PR does / why we need it:

* Add documentations around secret management in the datadog helm chart. It is to upstream 
  requested changes in the IBM charts repository: https://github.com/IBM/charts/pull/690#discussion_r411702458
* update `kube-state-metrics` dependency
* uncomment every values.yaml parameter for IBM chart compliancy

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

apigirl commented on stable/datadog/README.md:
```diff
@@ -31,6 +31,26 @@ helm install --name <RELEASE_NAME> \
 
+#### Create and provide a secret that contains your Datadog API Key
+
+To create a secret that contains your Datadog API Key Replace the <DATADOG_API_KEY> below with the API key for your organization. This secret is used in the manifest to deploy the Datadog Agent.
```
```suggestion
To create a secret that contains your Datadog API key, replace the <DATADOG_API_KEY> below with the API key for your organization. This secret is used in the manifest to deploy the Datadog Agent.
```

apigirl commented on stable/datadog/README.md:
````diff
@@ -31,6 +31,26 @@ helm install --name <RELEASE_NAME> \
+kubectl create secret generic $DATADOG_SECRET_NAME --from-literal api-key="<DATADOG_API_KEY>" --namespace="default"
+```
+
+**Note**: This create a secret in the default namespace. If you are in a custom namespace, update the namespace parameter of the command before running it.
````
```suggestion
**Note**: This creates a secret in the default namespace. If you are in a custom namespace, update the namespace parameter of the command before running it.
```

apigirl commented on stable/datadog/README.md:
````diff
@@ -31,6 +31,26 @@ helm install --name <RELEASE_NAME> \
+  --set datadog.apiKeyExistingSecret=$DATADOG_SECRET_NAME stable/datadog
+```
+
+**Note**: In the same way, it is possible to provides a secret for the Application Key (AppKey) thanks to the `datadog.appKeyExistingSecret` chart variable.
````
```suggestion
**Note**: Provide a secret for the application key (AppKey) using the `datadog.appKeyExistingSecret` chart variable.
```

apigirl reviewed (commented):
small feedback

clamoriniere committed:
[stable/datadog] ksm + doc update

* Add documentations around secret management in the datadog helm chart. It is to upstream
  requested changes in the IBM charts repository: https://github.com/IBM/charts/pull/690#discussion_r411702458
* update `kube-state-metrics` dependency
* uncomment every values.yaml parameter for IBM chart compliancy

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
[stable/datadog] update KSM dependency

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
[stable/datadog] update KSM dependency

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>
